spring-data-neo4j/src/main/java/org/springframework/data/neo4j/repository/query/filter/BetweenComparisonBuilder.java

```diff
@@ -39,22 +42,71 @@
 
 	@Override
 	public List<Filter> build(Stack<Object> params) {
-		final Object value1 = params.pop();
-		Filter gt = new Filter(propertyName(), ComparisonOperator.GREATER_THAN_EQUAL, value1);
-		gt.setOwnerEntityType(entityType);
-		gt.setBooleanOperator(booleanOperator);
-		gt.setNegated(isNegated());
-		gt.setFunction(new PropertyComparison(value1));
-		setNestedAttributes(part, gt);
-
-		final Object value2 = params.pop();
-		Filter lt = new Filter(propertyName(), ComparisonOperator.LESS_THAN_EQUAL, value2);
-		lt.setOwnerEntityType(entityType);
-		lt.setBooleanOperator(BooleanOperator.AND);
-		lt.setNegated(isNegated());
-		lt.setFunction(new PropertyComparison(value2));
-		setNestedAttributes(part, lt);
-
-		return Arrays.asList(gt, lt);
+
+		final Object lowerBoundOrRange = params.pop();
+
+		Object lowerBoundValue;
+		Object upperBoundValue;
+
+		boolean inclusiveLowerBound = true;
+		boolean inclusiveUpperBound = true;
+
+		if (lowerBoundOrRange instanceof Range) {
+			Range range = (Range) lowerBoundOrRange;
+			Range.Bound lowerBound = range.getLowerBound();
+			lowerBoundValue = lowerBound.getValue();
+			inclusiveLowerBound = lowerBound.isInclusive();
+
+			Range.Bound upperBound = range.getUpperBound();
+			upperBoundValue = upperBound.getValue();
+			inclusiveUpperBound = upperBound.isInclusive();
+		} else {
+			lowerBoundValue = lowerBoundOrRange;
+			upperBoundValue = params.pop();
+		}
+
+		Filter lowerBoundFilter = createLowerBoundFilter(lowerBoundValue, inclusiveLowerBound);
+		setNestedAttributes(part, lowerBoundFilter);
+
+		Filter upperBoundFilter = createUpperBoundFilter(upperBoundValue, inclusiveUpperBound);
+		setNestedAttributes(part, upperBoundFilter);
+
+		return Arrays.asList(lowerBoundFilter, upperBoundFilter);
+	}
+
+	private Filter createLowerBoundFilter(Object value, boolean inclusive) {
+		Filter filter = createBoundFilter(Bound.LOWER, value, inclusive);
+		filter.setBooleanOperator(booleanOperator);
```

Comment:
Can we pass on the operator on both of those (lower and upper)? 
The "and" is used for the boolean operator between previous and this or this and next?

Reply:
The "and" is for "greater/equals than lower bound AND lower/equals than upper bound". The `booleanOperator` (And/Or) is for the previous ones like `findByNameEqualsAnd/OrSomethingBetween`.